addons/mail/models/mail_message.py

```diff
@@ -981,13 +1011,14 @@ def _invalidate_documents(self, model=None, res_id=None):
             model = model or record.model
             res_id = res_id or record.res_id
             if issubclass(self.pool[model], self.pool['mail.thread']):
-                self.env[model].invalidate_cache(fnames=[
+                fnames = [
                     'message_ids',
-                    'message_unread',
-                    'message_unread_counter',
                     'message_needaction',
                     'message_needaction_counter',
-                ], ids=[res_id])
+                ]
+                if issubclass(self.pool[model], self.pool['mail.channel']):
```

Comment:
You should probably make a model (mail.thread maybe ?) method instead of hacking message model. Presence of model / res_id parameters indicates this method was located at the wrong place initially.